lib/collector_registration.rb

```diff
@@ -0,0 +1,139 @@
+require 'uri'
+
+require 'global_configuration'
+require 'json'
+require 'logging'
+require 'rest_client_extensions'
+
+class CollectorRegistration
+  include GlobalConfiguration
+  include Logging
+
+  def initialize
+    @environment = ENV['METER_ENV'] || 'development'
+    @configuration = GlobalConfiguration::GlobalConfig.instance
+  end
+
+  def configure_uc6
+    if credentials_present?
+      verify_connection_from_credentials
+    else #no credentials
+      if !@configuration.blank_value?(:uc6_api_host)
+        verify_connection_retrieving_missing_info
+      else
+        logger.error "Pending configuration of credentials"
+        exit(1)
+      end
+    end
+  end
+
+  private
+
+  def credentials_present?
+    !@configuration.blank_value?(:uc6_application_id) && !@configuration.blank_value?(:uc6_application_secret) &&
+    !@configuration.blank_value?(:uc6_login_email) && !@configuration.blank_value?(:uc6_login_password)
+  end
+
+  def verify_connection_from_credentials
+    if uc6_api_configured? && oauth_succesful?
+      retrieve_organization_name
+      @configuration[:verified_api_connection] = true
+      logger.info "Succesful integration with uc6 api"
+    else
+      exit(1)
+    end
+  end
+
+  def verify_connection_retrieving_missing_info
+    retrieve_organization_name
+    if !@configuration.blank_value?(:uc6_organization_name)
+      @configuration[:verified_api_connection] = true
+      logger.info "Succesful integration with uc6 api"
+    end
+  end
+
+  def uc6_api_configured?
+    !@configuration.blank_value?(:uc6_api_host) && is_valid_user?
+  end
+
+  def is_valid_user?
+    @configuration.blank_value?(:uc6_refresh_token) ?
+      validate_email & validate_password :
+      true
+  end
+
+  def validate_email
```

Comment:
maybe we could convert this method to a single line so that it returns directly the result of the condition statement, something like: 
`def validate_email
    !@configuration.blank_value?(:uc6_login_email) and @configuration[:uc6_login_email].match(/\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\Z/i) )
end`